### ISSUE TITLE
Reference model must use eval() and torch.inference_mode()

### ISSUE THREAD

**Issue by jashvira** [jashvira/policy-gradient-experiments#3]
Problem
- Inference/reference model is never set to eval().
- Old-policy log-probs and validation/generation are executed without torch.inference_mode().

Impact
- Dropout/other train-time layers active during rollouts -> unstable ratios in GRPO-Clip.
- Extra autograd overhead/memory on eval device.

Code
```python
# experiments/grpo/grpo_train.py
inference_model = init_inference_model_from(model, eval_device)
for p in inference_model.parameters(): p.requires_grad_(False)
# missing: inference_model.eval() and inference_mode() wrappers
```

Proposed fix
- Call inference_model.eval() immediately (before compile).
- Wrap rollouts, old-policy logprobs, and validation in torch.inference_mode().

Acceptance criteria
- Eval mode set and preserved.
- All inference paths use inference_mode().

**Comment by jashvira** [comment]
Fixed by f95788e: enforce eval()/inference_mode() for reference paths; preserve training mode.
